robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```diff
@@ -0,0 +1,1852 @@
+package org.obolibrary.robot.template;
+
+import com.google.common.collect.Lists;
+import java.util.*;
+import org.obolibrary.robot.IOHelper;
+import org.obolibrary.robot.QuotedEntityChecker;
+import org.semanticweb.owlapi.apibinding.OWLManager;
+import org.semanticweb.owlapi.manchestersyntax.parser.ManchesterOWLSyntaxClassExpressionParser;
+import org.semanticweb.owlapi.model.*;
+import org.semanticweb.owlapi.util.SimpleShortFormProvider;
+
+/** @author <a href="mailto:rctauber@gmail.com">Becky Tauber</a> */
+public class Template {
+
+  /** Template IOHelper to resolve prefixes. */
+  private IOHelper ioHelper;
+
+  /** Template QuotedEntityChecker to get entities and IRIs by label. */
+  private QuotedEntityChecker checker;
+
+  /** Manchester Syntax parser to parse class expressions. */
+  private ManchesterOWLSyntaxClassExpressionParser parser;
+
+  /** Set of axioms generated from template. */
+  private Set<OWLAxiom> axioms;
+
+  /** Location of IDs (ID). */
+  private int idColumn = -1;
+
+  /** Location of labels (LABEL, A rdfs:label, A label). */
+  private int labelColumn = -1;
+
+  /** Location of entity types (TYPE). */
+  private int typeColumn = -1;
+
+  /** Location of class types (CLASS_TYPE). */
+  private int classTypeColumn = -1;
+
+  /** Location of property types (PROPERTY_TYPE). */
+  private int propertyTypeColumn = -1;
+
+  /** Location of individual types (INDIVIDUAL_TYPE). */
+  private int individualTypeColumn = -1;
+
+  /** Character to split property types on. */
+  private String propertyTypeSplit = null;
+
+  /** Character to split individual types on. */
+  private String individualTypeSplit = null;
+
+  /** List of human-readable template headers. */
+  private List<String> headers;
+
+  /** List of ROBOT template strings. */
+  private List<String> templates;
+
+  /** All other rows of the table (does not include headers and template strings). */
+  private List<List<String>> tableRows;
+
+  /** Row number tracker. */
+  private int rowNum = 0;
+
+  /** Shared data factory. */
+  private final OWLDataFactory dataFactory = OWLManager.getOWLDataFactory();
+
+  /** Namespace for error messages. */
+  private static final String NS = "template#";
+
+  /**
+   * Error message when the number of header columns does not match the number of template columns.
+   * Expects: table name, header count, template count.
+   */
+  private static final String columnMismatchError =
+      NS
```

Comment:
Formatting problem?

Reply:
I think the next string was just too long and the auto-formatter did this weird thing. I'll divide it up better.